### DISCUSSION TITLE
Unexpected behavoir with Leaflet set_center and set_zoom

### DISCUSSION THREAD

**Discussion by thetableman** [zauberzeug/nicegui#2825] [Q&A]
Consider the below Leaflet example:
```py
from nicegui import ui

async def reset_location():
    m.set_center(center=(51.50, -0.11))
    m.set_zoom(11)
    print(m.center)
    print(m.zoom)

with ui.row().classes('h-96 w-96'):
    ui.button('Reset location', on_click=reset_location)
    m = ui.leaflet(center=(51.50, -0.11), zoom=11)
    print(m.center)
    print(m.zoom)

ui.run()
```

When executed it displays as expected.
![image](https://github.com/zauberzeug/nicegui/assets/78121198/ac3f4920-7095-4d3d-9334-722e8c0b6a2d)

However, when you move the map and change the zoom to something that resembles this...
![image](https://github.com/zauberzeug/nicegui/assets/78121198/f564a86a-fe2e-47d9-a656-fbba670ed212)

And then click `Reset Location`, the map doesn't display as expected, instead it just zooms in.
![image](https://github.com/zauberzeug/nicegui/assets/78121198/6ae044b8-7d62-40ac-8e17-3d4b2c0f958e)

Sometimes if you click `Reset Location` again it resets as expected. Other times it doesn't seem to matter how often you click `Reset Location`, the map doesn't display the expected location.

You'll also see inconsistencies in the way `m.center` returns a value
![image](https://github.com/zauberzeug/nicegui/assets/78121198/d61bf51f-2452-4f56-b354-390e52b6ac9c)

**Comment by falkoschindler** [comment]
Thanks for reporting this behavior, @thetableman!
I could reproduce the bug and created issue #2826 to track it. Let's continue the discussion over there.
